pkg/userdata/centos/userdata.go

```diff
@@ -94,6 +95,10 @@ func (p Provider) UserData(spec machinesv1alpha1.MachineSpec, kubeconfig *client
 		return "", fmt.Errorf("failed to get provider config: %v", err)
 	}
 
+	if pconfig.Network != nil {
```

Comment:
Ideally we would have a `Validate` for the user data providers, I will create an issue for that